### ISSUE TITLE
State of the project

### ISSUE THREAD

**Issue by herwix** [peerlibrary/peerlibrary#716]
Hey you all, 

just wanted to ask, whether you are still working on this project or if not, why not? It seemed like you had a very ambitious but awesome idea :) I am a phd student, frustrated with current tools for capturing, testing and sharing knowledge and I have some ideas of how we could make it better. Probably similar to you I don't have that much time to invest (doing too many things in parallel), but if people come together great things can be achieved. Are you still interested in the project or working on something similar? Maybe it would be interesting to talk if not for collaborating at least exchanging ideas :) 

Cheers, Alex

**Comment by mitar** [comment]
Currently it is in what I call the "zombie mode". It is still there, but nobody is really working on it. I am  really sad about that, but this is the reality of it.

I completely agree that it is very relevant and still needed. Even after few years I still haven't seen anything like it around.

There are many reasons which contributed to this. One is probably that we spend too much time on fighting some technical challenges (like PDF.js, Meteor). We were using them when they were still in early stages, so while this allowed us quick prototyping, it required much more effort to get to the production-grade which we wanted to achieve. Some our changes to PDF.js (like improved text selection) was just finally merged into upstream just few months ago.

We are/were mostly students, and students eventually graduate and move on. Also realities of my own PhD pressured me (I was unable to make PeerLibrary PhD itself, what I hoped). We didn't obtain [funding](https://github.com/peerlibrary/peerlibrary/issues/715#issuecomment-231820927) which would allow us to maybe operate independently. 

So yes, it is a very ambitious project.

Currently it is on a very old version of Meteor which really prevents any contributions, so [I tried to port it](https://github.com/peerlibrary/peerlibrary/tree/meteor0.9.0) but didn't have time to finish.

I am currently working [on another project](https://github.com/peer/mind) and I hope that many of underlying technologies I am developing there could be reused for PeerLibrary eventually, making it easier to rewrite the code for new Meteor.

So yes, I agree, we could work together. I would love that. But at this current stage I have no idea how to make this happen technically. It just requires somebody to sit down and port. I am probably the most suitable for that because I know the most of the codebase. But no time. :-( I hope to maybe come back to this after I finish my PhD. We will see.

We can definitely talk, but you can also just go around the issues and add your ideas here, so that they are not lost.

**Comment by herwix** [comment]
Hey Mitar, 

thank you for the quick and detailed response :)

I can very much relate to your sentiment, doing my phd as well and it really is difficult to keep everything on track (next to working part time). I hope that I could work on something similar to PeerLibrary as part of my phd though and maybe get a few students to write a master thesis on it as well. My focus would be a little bit different though... As a first step, I would want to focus on a decentralized blockchain app that allows researchers to keep track of the codings and concepts that they create in their qualitative data analysis (papers for in literature reviews or transcripts, etc.). The idea would be that people are able to privately store this data or make it public as they see fit.  

As it turns out meteor is a perfect fit for apps built on the ethereum block chain. So it might be possible to port some of the code. The parts relating to the annotation of pdf could be useful. Do you think that would be a good idea or should I rather start from scratch? I am a somewhat experienced developer but I really don't have much experience with meteor, pdfjs or your code base so any insight would be greatly appreciated :) 

Cheers, Alex

**Comment by mitar** [comment]
> I hope that I could work on something similar to PeerLibrary as part of my phd though and maybe get a few students to write a master thesis on it as well.

What are research contributions you think you can do? ;-) I have some ideas about that as well, but at the end it was still hard to sell it as a computer science novel research. Maybe one can find later on some smaller segments to do research on, when the system is up an running and you can collect some data, but for CS it was not a good match.

> As a first step, I would want to focus on a decentralized blockchain app 

Blockchain is really unnecessary here. You would make your life unnecessary harder, make app less easy to use, less responsive (waiting for data to be committed), you would be building on top of moving target and ecosystem. But the main question is: why. What is a threat model you are trying to address by using a decentralized storage system?

I think challenges for such apps are elsewhere and working on a decentralized app would just defocus you from those challenges.

[I wrote about this a bit](http://mitar.tnode.com/post/80878694215/proactive-federation-of-commons-is-better-than) in the past. You should maybe read few other posts on my blog about it.

You could check out the [beagle project](https://github.com/BeagleLab/voyage). They were working on a decentralized browser extension to discuss academic papers.

> Do you think that would be a good idea or should I rather start from scratch?

So current code is for old Meteor. It could probably be reused, but you anyway probably want to tightly couple your system with user experience of your system. I do not think you can really reuse code directly, but you could definitely look into how I am doing things.

For annotations, I would really suggest that you use [hypothes](https://hypothes.is/) annotator code, so that you stay compatible with the ecosystem, and store annotations using [open annotation standard](http://www.openannotation.org/). Also look into [ContentMine](http://contentmine.org/) for an example for coding. I think it would be much more useful to the whole ecosystem if you maybe build and work with others than trying to build something from scratch.

> Edit: P.S.: I reflected on your comment and project again and felt that I wanted to leave a more personal note. Maybe it is more for me because your project reminds me of my own, but maybe it is also helpful to you. It's dangerously philosophical but I guess that's how I am ;)
Your project reminded me of the incredible things we humans have achieved and will achieve - together. Just thinking about the knowledge and insights in all those papers written over generations and how a few people can come together to help make it accessible to each and every one of us is simply mesmerizing. And it is just so inspiring to see people like you trying to make a difference and stand in for what you deep in your heart believe in. I know it's a difficult balancing act that is different for every one. Don't beat yourself up over not completing everything you dreamed of or feel yourself pressured to do anything you don't want or have the time to. A project like yours can never be complete. Life or happiness in life is most of all a mindset. There is nothing you have to do - it's enough to simply be. Just know that people are out there who appreciate the work you have already done and are working on similar ideas... so if you have the time or simply feel like it there will always be a place for you :)

Thanks. I really appreciate it. I would just extend this to the whole PeerLibrary team, not just me. :-)

**Comment by herwix** [comment]
> What are research contributions you think you can do? ;-) I have some ideas about that as well, but at the end it was still hard to sell it as a computer science novel research. Maybe one can find later on some smaller segments to do research on, when the system is up an running and you can collect some data, but for CS it was not a good match.

I am not in CS but in information systems. I think it might be a little bit easier because we have a very  socio-technical perspective and are also interested in theories about how to support knowledge work effectively. I think it is interesting to take the case of a researcher... it's also close to my heart because all of the software I know does it wrong at the moment. That's actually what tipped me over to start working on this stuff now. I can test some theory that I have and scratch my own itch... sounds like a win, win to me :D 

> Blockchain is really unnecessary here. You would make your life unnecessary harder, make app less easy to use, less responsive (waiting for data to be committed), you would be building on top of moving target and ecosystem. But the main question is: why. What is a threat model you are trying to address by using a decentralized storage system?

> I think challenges for such apps are elsewhere and working on a decentralized app would just defocus you from those challenges.

Thank you for the input. I had some interesting applications in mind, however, you are certainly right it will make things a lot more difficult in the beginning. I am not dead set on building on the block chain at the moment and you have some very good points. Just one remark... it's not just about threat models but also affordances the block chain provides. Nevertheless, I will think about it again :)

> I wrote about this a bit in the past. You should maybe read few other posts on my blog about it.

> You could check out the beagle project. They were working on a decentralized browser extension to discuss academic papers.

> So current code is for old Meteor. It could probably be reused, but you anyway probably want to tightly couple your system with user experience of your system. I do not think you can really reuse code directly, but you could definitely look into how I am doing things.

> For annotations, I would really suggest that you use hypothes annotator code, so that you stay compatible with the ecosystem, and store annotations using open annotation standard. Also look into ContentMine for an example for coding. I think it would be much more useful to the whole ecosystem if you maybe build and work with others than trying to build something from scratch.

Thank you for the links and the pointers that will be very helpful and is really much appreciated! It's funny how I have had very similar ideas but never really came across these websites and projects... science is simply too disconnected! But very exciting stuff, maybe I will contact some of the projects and see if there is some interest in collaboration... I am very pragmatic in these things and just want to get some stuff done that's overdue :) 

> Thanks. I really appreciate it. I would just extend this to the whole PeerLibrary team, not just me. :-)

Of course. It was meant to be very inclusive :)

**Comment by mitar** [comment]
> I can test some theory that I have and scratch my own itch... sounds like a win, win to me :D 

That was my plan as well. :-) I am interested in massive online collaborative systems and technologies to support those. And it seems science could be one of main examples of such collaboration (instead competition).

> it's not just about threat models but also affordances the block chain provides

What do you mean here? Can you expand on this?

So I really think this is overhyped at the moment. So it is useful for some use cases, maybe, but you really have to be clear here what you are getting out of it. So what advantage do you think you would be getting from blockchain? What problem it is solving for you?

Because sorry, but I really have not seen many problems which you could not simply solve with simply distributing data by doing many copies. Mirrors, federations, Bittorrent. But check also [IPFS](http://ipfs.io/). I think what you maybe want is more IPFS, and not blockchain. Do you really care about double spend attack? Or you just want distributed permanent storage? :-) (And IPFS people are very into the open science.)

BTW, I would suggest you attend [OpenCon](http://www.opencon2016.org/).

> Thank you for the links and the pointers that will be very helpful and is really much appreciated! It's funny how I have had very similar ideas but never really came across these websites and projects... science is simply too disconnected!

Oh, you just didn't research enough. Let me help you. [Here is a long list, and not even complete, of various projects we found about two years ago](https://github.com/peerlibrary/peerlibrary/wiki/Related-Projects). Enjoy going through them. Feel free to edit wiki with descriptions of them. :-)

> I am very pragmatic in these things and just want to get some stuff done that's overdue :) 

So what exactly is your idea? :-) Because maybe I can connect you with some people.

**Comment by herwix** [comment]
> That was my plan as well. :-) I am interested in massive online collaborative systems and technologies to support those. And it seems science could be one of main examples of such collaboration (instead competition).

Maybe I misunderstood you but it sounds like you think of competition as a bad thing... From my point of view competition is not really a bad thing by itself... it's natural. We might not want to admit it but basically everything we do is for us. I think that will be one of the key ideas to make something like you envision work on a large scale.

> What do you mean here? Can you expand on this?

At the moment that is still difficult because I am mostly thinking about this stuff in very abstract terms and did not really have the time to evaluate the technology aspects in detail. What I can tell you is that I really like the blockchain concept because it seems to align well with the perspective I am developing. It solved a very interesting incentive problem and I think providing incentive will be key if the goal is to bring substantial change. I did not know about IPFS, though. It looks very interesting and might be a plausible alternative.

I hope to "open source" some of my thinking soon (What do you think of the OSF? Do you have a better place to publish work in progress type stuff?), right now I am basically flooded with research and work, though. 

> BTW, I would suggest you attend OpenCon.

Thanks for the pointer! Will try to make it online :)

> Oh, you just didn't research enough. Let me help you. Here is a long list, and not even complete, of various projects we found about two years ago. Enjoy going through them. Feel free to edit wiki with descriptions of them. :-)

Thanks, again! :) That's really helpful 👍  

> So what exactly is your idea? :-) Because maybe I can connect you with some people.

So my idea is more like a world view. I try to view the world through a very process oriented lens. To me it's very interesting from a philosophical but also from an action oriented point of view. I think I have a good way to conceptualize and make sense of processes that could be helpful to design 'such a system' in a way that it actually collects the right kind of data and can deliver the necessary incentives to drive adoption. It's nothing super fancy but I still need some time to reflect and put some words to it... especially in science people want to get everything on a silver platter and not think for themselves... gotta love the irony :)
